source/air/src/ast.rs

```diff
@@ -202,13 +202,23 @@ pub struct QueryX {
     pub assertion: Stmt, // checked by SMT with global and local declarations
 }
 
+pub type SingularQuery = Arc<SingularQueryX>;
+#[derive(Debug)]
+// for Singular queries only
+pub struct SingularQueryX {
+    pub local: Decls,
+    pub requires: Stmts,
+    pub ensures: Stmts,
+}
+
 pub type Command = Arc<CommandX>;
 pub type Commands = Arc<Vec<Command>>;
 #[derive(Debug)]
 pub enum CommandX {
-    Push,                    // push space for temporary global declarations
-    Pop,                     // pop temporary global declarations
-    SetOption(Ident, Ident), // set-option option value (no colon on the option)
-    Global(Decl),            // global declarations
-    CheckValid(Query), // SMT check-sat (reporting validity rather than satisfiability), Possibly singular checks
+    Push,                         // push space for temporary global declarations
+    Pop,                          // pop temporary global declarations
+    SetOption(Ident, Ident),      // set-option option value (no colon on the option)
+    Global(Decl),                 // global declarations
+    CheckValid(Query),            // SMT check-sat (reporting validity rather than satisfiability)
+    CheckSingular(SingularQuery), // Singular check
```

Comment:
It would be good, but not critical, if you could make the existence of this variant dependent on whether the `singular` feature is enabled. (with `#[cfg(feature = "singular")]`).
The same applies to the code in other files that handle it.

Reply:
I think that is a good point. Given that there is some discussion of lean or cvc5 integration, and potentially lean's ring tactics or cvc5's finite field solver may supersede Singular, so it is a good practice to put the current code under feature flag. I have made some changes accordingly.